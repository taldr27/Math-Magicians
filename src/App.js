import React from 'react';
import {
  Routes,
  Route,
} from 'react-router-dom';
import Home from './pages/Home';
import Navbar from './components/NavBar';
import Quote from './pages/Quote';
import Calculator from './pages/calculator';

function App() {
  return (
    <div>
      <div className="header">
        <h1>Math Magicians</h1>
        <Navbar />
      </div>
      <Routes>
        <Route path="/calculator" element={<Calculator />} />
        <Route path="/" element={<Home />} />
        <Route path="/quote" element={<Quote />} />
      </Routes>
    </div>
  );
}

export default App;
