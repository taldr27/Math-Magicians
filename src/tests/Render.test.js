import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import App from '../App';
import Navbar from '../components/NavBar';

test('Render Home and check quote text', async () => {
  render(<App />, { wrapper: BrowserRouter });
  const user = userEvent.setup();

  expect(screen.getByText(/calculator/i)).toBeInTheDocument();

  await user.click(screen.getByText(/quote/i));
  expect(screen.getByText(/- William Paul Thurston/i)).toBeInTheDocument();
});

test('Render NavBar', () => {
  const renderNavBar = renderer.create(<BrowserRouter><Navbar /></BrowserRouter>).toJSON();
  expect(renderNavBar).toMatchSnapshot();
});
