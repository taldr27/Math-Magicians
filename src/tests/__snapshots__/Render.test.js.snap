// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render NavBar 1`] = `
<nav>
  <ul
    className="navBar"
  >
    <li>
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;
